apps/internal/base/internal/storage/items_test.go

```diff
@@ -54,14 +55,16 @@ func TestCreateAccessToken(t *testing.T) {
 		testExtExpiresOn,
 		"user.read",
 		"access",
+		"tokenType",
+		"keyID",
 	)
 	if !extExpiresOn.Equal(actualAt.ExtendedExpiresOn.T) {
 		t.Errorf("Actual ext expires on %s differs from expected ext expires on %s", actualAt.ExtendedExpiresOn, extExpiresOn)
 	}
 }
 
 func TestKeyForAccessToken(t *testing.T) {
-	const want = "testhid-env-accesstoken-clientid-realm-user.read"
+	const want = "testhid-env-accesstoken-clientid-realm-user.read-bearer"
```

Comment:
I am concerned that this will break backwards  and forwards compatibility with the token cache. It's ok if older MSALs can't read POP tokens, but they should continue to read bearer tokens. As such, when the auth scheme is beaerr, we should leave the key alone. 

Here's the equivalent logic in MSAL.NET https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/AuthScheme/AuthSchemeHelper.cs

We should have some tests on this: 

- old cache format + new MSAL
- new cache format (i.e. with POP and Bearer tokens )+ old MSAL (I guess this will have to be manual test)

Reply:
Thanks. Fixed how key is generated in items.go. Updated test.

Reply:
Where can we find steps to to simulate
   _new cache format (i.e. with POP and Bearer tokens )+ old MSAL (I guess this will have to be manual test)_